src/app/datagrid/pagination-conditional/pagination-conditional.html

```diff
@@ -0,0 +1,40 @@
+<!--
+  ~ Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Conditional Pagination</h2>
+
+<p>
+    Test to conditional activate/deactivate pagination in a datagrid
+</p>
+
+<clr-datagrid style="max-height: 278px">
+    <clr-dg-column>User ID</clr-dg-column>
+    <clr-dg-column>Name</clr-dg-column>
+    <clr-dg-column>Creation date</clr-dg-column>
+    <clr-dg-column>Favorite color</clr-dg-column>
+
+    <clr-dg-row *clrDgItems="let user of users">
+        <clr-dg-cell>{{user.id}}</clr-dg-cell>
+        <clr-dg-cell>{{user.name}}</clr-dg-cell>
+        <clr-dg-cell>{{user.creation | date}}</clr-dg-cell>
+        <clr-dg-cell>
+            <span class="color-square" [style.backgroundColor]="user.color"></span>
+        </clr-dg-cell>
+    </clr-dg-row>
+
+    <clr-dg-footer>
+        <span *ngIf="pagination">
```

Comment:
Couldn't we include all the contents of `clr-dg-footer` in one `ng-container` and toggle it with single `paginationEnabled `?

Reply:
But all the contents in `clr-dg-footer` are not hidden when the pagination is disabled. `users.length` is still visible.

Reply:
I see. But we could still use only `paginationEnabled` in the place of `*ngIf="pagination"` and don't need to listen to `this.paginationViewChildren.changes` to set the `pagination` property to `null`.

Reply:
I believe we do need to listen to the view children changes because thats exactly when i need to run cdr.detectChanges() to void the chocolate error. If the pagination is not enabled, i dont understand why we do not need to set the pagination property to null. it wont reflect whats on the view if we dont set it to null.

this will be more clear when we know if we need the change detector ref or not. i will wait for eudes to respond.

Reply:
I might be wrong, but I believe doing something as simple as putting this span _after_ the pagination will fix your chocolate error. And that means you'd just use `paginationEnabled` like @Shijir suggested, and you can switch from the whole `@ViewChildren` + changes listener to a simple `@ViewChild` without any listeners.

Reply:
@youdz I already tried that before submitting the PR :D Didn't work :-(

Reply:
plnkr which replicates this: https://plnkr.co/edit/BMCrLP6AvFG6DWdhErB1?p=preview